pkg/controller/jobs/deployment/deployment_webhook.go

```diff
@@ -58,6 +64,23 @@ func (wh *Webhook) Default(ctx context.Context, obj runtime.Object) error {
 	log := ctrl.LoggerFrom(ctx).WithName("deployment-webhook")
 	log.V(5).Info("Applying defaults")
 
+	queue, ok := deployment.Labels[constants.QueueLabel]
+	if ok || features.Enabled(features.LocalQueueDefaulting) && wh.queues.DefaultLocalQueue(deployment.ObjectMeta.Namespace) != "" {
+		if deployment.Spec.Template.Labels == nil {
+			deployment.Spec.Template.Labels = make(map[string]string, 1)
+		}
+	}
+	if ok {
+		deployment.Spec.Template.Labels[constants.QueueLabel] = queue
+	} else if features.Enabled(features.LocalQueueDefaulting) && wh.queues.DefaultLocalQueue(deployment.ObjectMeta.Namespace) != "" {
+		queue := kueue_constants.DefaultLocalQueueName
+		if deployment.Labels == nil {
+			deployment.Labels = make(map[string]string)
+		}
+		deployment.Labels[constants.QueueLabel] = queue
+		deployment.Spec.Template.Labels[constants.QueueLabel] = queue
```

Comment:
We already have code which copies the label down to PodTemplate (below), so I think it would be great to reuse the ApplyDefaultLocalQueue function and don't duplicate the logic. Is there any blocker for that?